lib/custom/src/MShop/Service/Provider/Payment/PaypalPlus.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+/**
+ * @license LGPLv3, http://opensource.org/licenses/LGPL-3.0
+ * @copyright Aimeos (aimeos.org), 2015-2020
+ * @package MShop
+ * @subpackage Service
+ */
+
+
+namespace Aimeos\MShop\Service\Provider\Payment;
+
+
+use Omnipay\Omnipay as OPay;
+use Aimeos\MShop\Order\Item\Base as Status;
+
+
+/**
+ * Payment provider for payment gateways supported by the PaypalPlus library.
+ *
+ * @package MShop
+ * @subpackage Service
+ */
+class PaypalPlus
+extends    \Aimeos\MShop\Service\Provider\Payment\OmniPay
+//	extends \Aimeos\MShop\Service\Provider\Payment\Base
+	implements \Aimeos\MShop\Service\Provider\Payment\Iface
+{
+	private $beConfig = array(
+		'type' => array(
+			'code' => 'type',
+			'internalcode'=> 'type',
+			'label'=> 'Payment provider type',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '',
+			'required'=> true,
+		),
+		'onsite' => array(
+			'code' => 'onsite',
+			'internalcode'=> 'onsite',
+			'label'=> 'Collect data locally',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> false,
+		),
+		'address' => array(
+			'code' => 'address',
+			'internalcode'=> 'address',
+			'label'=> 'Send address to payment gateway too',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> false,
+		),
+		'authorize' => array(
+			'code' => 'authorize',
+			'internalcode'=> 'authorize',
+			'label'=> 'Authorize payments and capture later',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> false,
+		),
+		'createtoken' => array(
+			'code' => 'createtoken',
+			'internalcode'=> 'createtoken',
+			'label'=> 'Request token for recurring payments',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '1',
+			'required'=> false,
+		),
+		'testmode' => array(
+			'code' => 'testmode',
+			'internalcode'=> 'testmode',
+			'label'=> 'Test mode without payments',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> false,
+		),
+	);
+
+	private $feConfig = array(
```

Comment:
There's no form elements rendered by Aimeos, only by the PayPal JS code, isn't it? If yes, you can remove all those definitions

Reply:
thanks for note ,I have updated my code